### ISSUE TITLE
I am passing image with mask but i am not getting good results.

### ISSUE THREAD

**Issue by abhishek7467** [dvlab-research/MAT#1]
```
import cv2
import pyspng
import glob
import os
import re
import random
from typing import List, Optional
import click
import dnnlib
import numpy as np
import PIL.Image
import torch
import torch.nn.functional as F
```
```

import legacy
from datasets.mask_generator_512 import RandomMask
from networks.mat import Generator
os.environ['TORCH_CUDA_ARCH_LIST']
```
```

def num_range(s: str) -> List[int]:
    '''Accept either a comma separated list of numbers 'a,b,c' or a range 'a-c' and return as a list of ints.'''

    range_re = re.compile(r'^(\d+)-(\d+)$')
    m = range_re.match(s)
    if m:
        return list(range(int(m.group(1)), int(m.group(2))+1))
    vals = s.split(',')
    return [int(x) for x in vals]
```

```

def copy_params_and_buffers(src_module, dst_module, require_all=False):
    assert isinstance(src_module, torch.nn.Module)
    assert isinstance(dst_module, torch.nn.Module)
    src_tensors = {name: tensor for name, tensor in named_params_and_buffers(src_module)}
    for name, tensor in named_params_and_buffers(dst_module):
        assert (name in src_tensors) or (not require_all)
        if name in src_tensors:
            tensor.copy_(src_tensors[name].detach()).requires_grad_(tensor.requires_grad)


```
```
def params_and_buffers(module):
    assert isinstance(module, torch.nn.Module)
    return list(module.parameters()) + list(module.buffers())


```
```
def named_params_and_buffers(module):
    assert isinstance(module, torch.nn.Module)
    return list(module.named_parameters()) + list(module.named_buffers())
```

```

@click.command()
@click.pass_context
@click.option('--network', 'network_pkl', help='Network pickle filename', required=True)
@click.option('--dpath', help='the path of the input image', required=True)
@click.option('--mpath', help='the path of the mask')
@click.option('--resolution', type=int, help='resolution of input image', default=512, show_default=True)
@click.option('--trunc', 'truncation_psi', type=float, help='Truncation psi', default=1, show_default=True)
@click.option('--noise-mode', help='Noise mode', type=click.Choice(['const', 'random', 'none']), default='none', show_default=True)
 @click.option('--outdir', help='Where to save the output images', type=str, required=True, metavar='DIR')

```
```
def generate_images(
    ctx: click.Context,
    network_pkl: str,
    dpath: str,
    mpath: Optional[str],
    resolution: int,
    truncation_psi: float,
    noise_mode: str,
     outdir: str,
):
  
    seed = 240  # pick up a random number
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)

    # print(f'Loading data from: {dpath}')
    # img_list = sorted(glob.glob(dpath + '/*.png') + glob.glob(dpath + '/*.jpg'))

    # if mpath is not None:
    #     print(f'Loading mask from: {mpath}')
    #     mask_list = sorted(glob.glob(mpath + '/*.png') + glob.glob(mpath + '/*.jpg'))
    #     assert len(img_list) == len(mask_list), 'illegal mapping'
    network_pkl = "E:\\MAT_Inpainting\\MAT\\pretrained\\Places_512_FullData.pkl"

    img_path =  "E:\\Image_Inpainting\\ImageWebApp\\ImageApp\\imageForMat.png" 
    mpath =  "E:\\Image_Inpainting\\ImageWebApp\\ImageApp\\maskForMat.png"


    device = torch.device('cuda')
    with dnnlib.util.open_url(network_pkl) as f:
        G_saved = legacy.load_network_pkl(f)['G_ema'].to(device).eval().requires_grad_(False) # type: ignore
    
    net_res = 512 if resolution > 512 else resolution
    G = Generator(z_dim=512, c_dim=0, w_dim=512, img_resolution=net_res, img_channels=3).to(device).eval().requires_grad_(False)
    copy_params_and_buffers(G_saved, G, require_all=True)
    print("mks")
    # os.makedirs(outdir, exist_ok=True)

    # no Labels.
    label = torch.zeros([1, G.c_dim], device=device)
    print("label ", label)
    def read_image(image_path):
        with open(image_path, 'rb') as f:
            if pyspng is not None and image_path.endswith('.png'):
                image = pyspng.load(f.read())
            else:
                image = np.array(PIL.Image.open(f))
        if image.ndim == 2:
            image = image[:, :, np.newaxis] # HW => HWC
            image = np.repeat(image, 3, axis=2)
        image = image.transpose(2, 0, 1) # HWC => CHW
        image = image[:3]
        return image

    def to_image(image, lo, hi):
        image = np.asarray(image, dtype=np.float32)
        image = (image - lo) * (255 / (hi - lo))
        image = np.rint(image).clip(0, 255).astype(np.uint8)
        image = np.transpose(image, (1, 2, 0))
        if image.shape[2] == 1:
            image = np.repeat(image, 3, axis=2)
        return image
    print("resolution  ", resolution)
    if resolution != 512:
        noise_mode = 'random'
    with torch.no_grad():


       
        image = read_image(img_path)
        image = (torch.from_numpy(image).float().to(device) / 127.5 - 1).unsqueeze(0)

       
        mask = cv2.imread(mpath, cv2.IMREAD_GRAYSCALE).astype(np.float32) / 255.0
        mask = torch.from_numpy(mask).float().to(device).unsqueeze(0).unsqueeze(0)
        print("mpathmpathmpathmpathmpathmpathmpathmpathmpathmpathmpathmpath")
        



        z = torch.from_numpy(np.random.randn(1, G.z_dim)).to(device)

        print("z --->  ", z)
        print("label --->  ", label)
        print("truncation_psi --->  ", truncation_psi)
        print("noise_mode --->  ", noise_mode)
        output = G(image, mask, z, label, truncation_psi=truncation_psi, noise_mode=noise_mode)
        output = (output.permute(0, 2, 3, 1) * 127.5 + 127.5).round().clamp(0, 255).to(torch.uint8)
        output = output[0].cpu().numpy()
        print("Completed!!!")
        cv2.imwrite("output.png",output)
```

```

if __name__ == "__main__":
    generate_images() # pylint: disable=no-value-for-parameter
```

#----------------------------------------------------------------------------

I am building an application in which I want to remove a particular object from an image based on its mask.
I have tried the LAMA + Refinement feature but I am not satisfied with Its results.
I have modified the above code. You can see here that I passed the direct path of the image and the mask after completely executing the code.

I have attached my Input and output Images below: 
You can see my input image which is in 512*512 ![image](https://github.com/user-attachments/assets/b4c48e7e-af4b-4282-b98c-92cd5afa67cb)
You can see my input mask which is in 512*512  ![mask](https://github.com/user-attachments/assets/74d1ac74-2fc0-4784-ab3f-4df6065e76ca)
  But I am getting this result which is not expected from It.     
  
  
![output](https://github.com/user-attachments/assets/3075d2de-b841-4f96-8130-f4911363c391)

  
  but I like it will inpaint only masked object in the image. not should disturb the entire image

**Comment by abhishek7467** [comment]
I want output like the below result this is the output of the LAMA Inpainting in which you can see only the target is marked output in the image. It is not disturbing other objects.


![Screenshot 2024-12-09 191331](https://github.com/user-attachments/assets/478bb1e2-cd22-4834-a098-e0624c6f38b0)
